provider/pkg/schema/gen.go

```diff
@@ -628,6 +644,42 @@ func (ctx *context) gatherResourceType() error {
 			requiredInputs.Add(sdkName)
 		}
 	}
+
+	// ** Autonaming **
+	autonameSpecs := map[string]AutoNamingSpec{}
+
+	lookForField := func(fieldName string) func(map[string]pschema.PropertySpec) {
+		return func(properties map[string]pschema.PropertySpec) {
+			sdkName := ToSdkName(fieldName)
+			if propSpec, has := inputProperties[sdkName]; has && propSpec.Type == "string" {
+				autonameSpec := AutoNamingSpec{}
+				spec, ok := ctx.resourceSpec.Properties[fieldName]
+				if ok {
+					autonameSpec.MinLength = spec.MinLength.Val
+					autonameSpec.MaxLength = spec.MaxLength.Val
+				}
+				autonameSpecs[sdkName] = autonameSpec
+			}
+		}
+	}
+
+	autoNamePropLookupFuncs := []func(map[string]pschema.PropertySpec){
+		lookForField("Name"),
+		lookForField(resourceTypeName+"Name"),
+	}
+
+	for _, fn := range autoNamePropLookupFuncs {
+		fn(properties)
+		if len(autonameSpecs) > 0 {
+			for autoNamed := range autonameSpecs {
+				// Apply auto-naming, i.e. mark the property as no longer required.
+				delete(requiredInputs, autoNamed)
+			}
+			break
+		}
+	}
+	// ** Autonaming **
```

Comment:
May be worth refactoring into a function instead of the commented block?

Reply:
Done.